### PR TITLE
Photo import REST API and service

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

File: user-data-store-server/src/main/java/com/wultra/security/userdatastore/controller/PhotoController.java [hunk 1]
```diff
@@ -123,4 +126,40 @@ public Response deletePhotos(@NotBlank @Size(max = 255) @RequestParam String use
...
+     * Import photos asynchronously from CSV.
+     *
+     * @param request Photo import from CSV request
+     * @return Response
+     */
+    @Operation(
+            summary = "Import photos from CSV",
+            description = "Import photos asynchronously from CSV."
+    )
+    @PostMapping("/admin/photos/import/csv")
```
Comment (banterCZ) [inline user-data-store-server/src/main/java/com/wultra/security/userdatastore/controller/PhotoController.java:157]: 🤔 A nitpick: I was about to suggest extracting the common path into `@RequestMapping` but I see that we mix admin and non-admin context. Shouldn't we split the controller into two classes?

File: user-data-store-server/src/main/java/com/wultra/security/userdatastore/model/entity/ImportResultEntity.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    @Column(name = "error")
+    private String error;
+
+    @Column(name = "timestamp_created", nullable = false)
+    private LocalDateTime timestampCreated = LocalDateTime.now();
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```
Comment (banterCZ) [inline user-data-store-server/src/main/java/com/wultra/security/userdatastore/model/entity/ImportResultEntity.java:74]: Shouldn't we prefer `ProxyUtils`? https://github.com/wultra/development-internal/wiki/Equals-and-hashCode-in-JPA-entities

File: user-data-store-server/src/main/java/com/wultra/security/userdatastore/service/AttachmentService.java [hunk 3]
```diff
@@ -146,7 +146,7 @@ public void deleteAttachments(final String userId, final Optional<String> docume
     }
 
     private void audit(final String message, final String userId, final String documentId) {
-        final String loggedUsername = SecurityContextHolder.getContext().getAuthentication().getName();
+        final String loggedUsername = SecurityContextHolder.getContext().getAuthentication() != null ? SecurityContextHolder.getContext().getAuthentication().getName() : null;
```
Comment (banterCZ) [inline user-data-store-server/src/main/java/com/wultra/security/userdatastore/service/AttachmentService.java:149]: Just an idea
```suggestion
        final String loggedUsername = Optional.ofNullable(SecurityContextHolder.getContext().getAuthentication())
                .map(Authentication::getName)
                .orElse(null);
```

File: user-data-store-server/src/test/java/com/wultra/security/userdatastore/restclient/PhotoRestClientTest.java [hunk 4]
```diff
@@ -136,4 +140,221 @@ void testValidation_InvalidBase64() {
...
+    private void verifyImportCsv(List<String> userIds, String expectedPhotoBase64) {
+        userIds.forEach(userId -> {
+            for (int i = 0; i < 10; i++) {
+                assertNotEquals(9, i);
+                try {
+                    verifyImportCsv(userId, expectedPhotoBase64);
+                    break;
+                } catch (Exception ex) {
+                    try {
+                        Thread.sleep(100);
```
Comment (banterCZ) [inline user-data-store-server/src/test/java/com/wultra/security/userdatastore/restclient/PhotoRestClientTest.java:344]: May we use `Awaitility` instead?

**Review by banterCZ — APPROVED** [review]
Looks good. I have added several suggestions to consider.

File: user-data-store-server/src/main/java/com/wultra/security/userdatastore/controller/PhotoController.java (hunk 1 above)
Comment (romanstrobl) [reply user-data-store-server/src/main/java/com/wultra/security/userdatastore/controller/PhotoController.java:157]: I prefer to group methods related to the same functionality in one controller class.

File: user-data-store-server/src/main/java/com/wultra/security/userdatastore/model/entity/ImportResultEntity.java (hunk 2 above)
Comment (romanstrobl) [reply user-data-store-server/src/main/java/com/wultra/security/userdatastore/model/entity/ImportResultEntity.java:74]: Ah, yes, the code was generated.

**Pull request merged**